flambe/metric/metric.py

```diff
@@ -32,10 +45,58 @@ def compute(self, pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
         """
         pass
 
+    def aggregate(self, state: dict, *args, **kwargs) -> Dict:
+        """
+
+        Parameters
+        ----------
+        state: dict
+            the state dictionary
+        args:
+            normally pred, target
+        kwargs
+
+        Returns
+        -------
+        The updated state (even though the update happens in-place)
+        """
+        score = self.compute(*args, **kwargs)
+        if isinstance(score, torch.Tensor):
+            score_np = score.cpu().detach().numpy()
+        else:
+            score_np = score
+        try:
+            num_samples = args[0].size(0)
+        except (ValueError, AttributeError):
+            raise ValueError('If the first arg to the metric function\'s incremental method is not a tensor,'
+                             'provide the num_samples kwarg')
+        if state == {}:
+            state['accumulated_score'] = 0.
+            state['sample_count'] = 0
+        state['accumulated_score'] = (state['sample_count'] * state['accumulated_score'] +
+                                      num_samples * score_np.item()) / (state['sample_count'] + num_samples)
+        state['sample_count'] = state['sample_count'] + num_samples
+        return state
+
+    def finalize(self, state) -> float:
+        """
+        FInalizes the metric computation
+        
+        Parameters
+        ----------
+        state: dict
+            the metric state
+
+        Returns
+        -------
+        The final score
+        """
+        return state['accumulated_score'] if 'accumulated_score' in state else None
+
     def __call__(self, *args, **kwargs):
         """Makes Featurizer a callable."""
         return self.compute(*args, **kwargs)
 
     def __str__(self) -> str:
         """Return the name of the Metric (for use in logging)."""
-        return self.__class__.__name__
+        return self.__class__.__name__ if getattr(self, 'name', None) is None else self.name
```

Comment:
calling getattr is weird here. Just set self.name to None in the parent __init__ and use self.name is None here